### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by pkanwar23**
Updating file to reflect our ability to take contributions.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,9 +1,73 @@
 # How to Contribute
 
-We'd love to accept your patches and contributions to this project soon. But we
-are not yet ready to accept community contributions at this time.
+Everyone is welcome to contribute to MLIR. There are several ways of getting involved and contributing including reporting bugs, improving documentation, writing models or tutorials. 
+
+Please read our our [Code of Conduct](https://github.com/tensorflow/swift/blob/master/CODE_OF_CONDUCT.md) before participating.
```
Comment (antiagainst) [inline CONTRIBUTING.md]: Should we check a copy here instead of referencing to another repo?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,9 +1,73 @@
...
+***NOTE***: Only original source code from you and other people that have signed the CLA can be accepted into the main repository.
+
+### Contributing code
+
+If you have improvements to TensorFlow, send us your pull requests! For those
+just getting started, Github has a [howto](https://help.github.com/articles/using-pull-requests/).
+
+MLIR team members will be assigned to review your pull requests. Once the pull requests are approved and pass continuous integration checks, a team member will merge your pull request submitted to our internal repository. After the change has been submitted internally, your pull request will be merged automatically on GitHub.
+
+If you want to contribute, start working through the TensorFlow codebase, navigate to [Github "issues" tab](https://github.com/tensorflow/mlir/issues) and start looking through interesting issues. If you decide to start on an issue, leave a comment so that other people know that you're working on it. If you want to help out, but not alone, use the issue comment thread to coordinate.
```
Comment (antiagainst) [inline CONTRIBUTING.md]: MLIR codebase ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,9 +1,73 @@
...
+#### License
+
+Include a license at the top of new files.
+
+* [C/C++ license example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/op.cc#L1)
+* [Python license example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/nn.py#L1)
+
+#### C++ coding style
+
+Changes to TensorFlow C++ code should conform to
```
Comment (antiagainst) [inline CONTRIBUTING.md]: MLIR C++ code?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,9 +1,73 @@
...
+[LLVM Coding standards](https://llvm.org/docs/CodingStandards.html).
+
+#### Python coding style
+
+Changes to TensorFlow Python code should conform to
```
Comment (antiagainst) [inline CONTRIBUTING.md]: MLIR Python code?

File: CONTRIBUTING.md (hunk 2 above)
Comment (joker-eph) [inline CONTRIBUTING.md]: "working through the TensorFlow codebase" => "working through the MLIR codebase"

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,9 +1,73 @@
...
+    changes to lower the maintenance cost.
+*   Bug fixes also generally require unit tests, because the presence of bugs
+    usually indicates insufficient test coverage.
+
+#### License
+
+Include a license at the top of new files.
+
+* [C/C++ license example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/op.cc#L1)
+* [Python license example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/nn.py#L1)
```
Comment (joker-eph) [inline CONTRIBUTING.md]: Can you update with links to the MLIR source code?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,9 +1,73 @@
...
+
+MLIR team members will be assigned to review your pull requests. Once the pull requests are approved and pass continuous integration checks, a team member will merge your pull request submitted to our internal repository. After the change has been submitted internally, your pull request will be merged automatically on GitHub.
+
+If you want to contribute, start working through the TensorFlow codebase, navigate to [Github "issues" tab](https://github.com/tensorflow/mlir/issues) and start looking through interesting issues. If you decide to start on an issue, leave a comment so that other people know that you're working on it. If you want to help out, but not alone, use the issue comment thread to coordinate.
+
+### Contribution guidelines and standards
+
+*   Follow coding style (see below).
+*   Ensure that you use the correct license. Examples are provided below.
+*   Include unit tests when you contribute new features, as they help to a)
```
Comment (joker-eph) [inline CONTRIBUTING.md]: "Include unit tests" => "Include tests"

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,9 +1,73 @@
...
+Changes to TensorFlow C++ code should conform to
+[LLVM Coding standards](https://llvm.org/docs/CodingStandards.html).
+
+#### Python coding style
+
+Changes to TensorFlow Python code should conform to
+[Google Python Style Guide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md)
+
+Use `pylint` to check your Python changes. To install `pylint` and
+retrieve TensorFlow's custom style definition:
```
Comment (joker-eph) [inline CONTRIBUTING.md]: You can remove this section, there is not Python style guide in LLVM (AFAIK)

File: CONTRIBUTING.md (hunk 1 above)
Comment (pkanwar23) [reply CONTRIBUTING.md]: I've updated to point to TF. I don't want to create another copy, else we'll run the risk of getting out of sync.

File: CONTRIBUTING.md (hunk 2 above)
Comment (pkanwar23) [reply CONTRIBUTING.md]: done.

File: CONTRIBUTING.md (hunk 4 above)
Comment (pkanwar23) [reply CONTRIBUTING.md]: done.

File: CONTRIBUTING.md (hunk 3 above)
Comment (pkanwar23) [reply CONTRIBUTING.md]: done.

File: CONTRIBUTING.md (hunk 2 above)
Comment (pkanwar23) [reply CONTRIBUTING.md]: done.

File: CONTRIBUTING.md (hunk 5 above)
Comment (pkanwar23) [reply CONTRIBUTING.md]: done.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,9 +1,54 @@
 # How to Contribute
 
-We'd love to accept your patches and contributions to this project soon. But we
-are not yet ready to accept community contributions at this time.
+Everyone is welcome to contribute to MLIR. There are several ways of getting involved and contributing including reporting bugs, improving documentation, writing models or tutorials. 
+
+Please read our our [Code of Conduct](https://github.com/tensorflow/tensorflow/blob/master/CODE_OF_CONDUCT.md) before participating.
```
Comment (joker-eph) [inline CONTRIBUTING.md]:
```suggestion
Please read our [Code of Conduct](https://github.com/tensorflow/tensorflow/blob/master/CODE_OF_CONDUCT.md) before participating.
```

**Pull request merged**